packages/cli/src/utils/analytics.js

```diff
@@ -15,19 +15,22 @@ const setAnalyticsMode = newMode => {
   return writeUserConfig({ [ANALYTICS_KEY]: newMode });
 };
 
+const shouldSkipAnalytics = mode =>
+  IS_TESTING ||
+  process.env.DISABLE_ZAPIER_ANALYTICS ||
+  mode === ANALYTICS_MODES.disabled;
+
 const recordAnalytics = async (command, isValidCommand, args, flags) => {
   const analyticsMode = await currentAnalyticsMode();
 
-  const shouldRecordAnalytics =
-    process.env.DISABLE_ZAPIER_ANALYTICS ||
-    (process.NODE_ENV !== 'test' && analyticsMode !== ANALYTICS_MODES.disabled);
```

Comment:
❗️ the actual bug was having `process.NODE_ENV` instead of `process.env.NODE_ENV`